messages/events/dailydebitlimit.go

```diff
@@ -53,3 +55,57 @@ func (m DailyDebitLimitExceeded) MessageDescription() string {
 		messages.FormatAmount((m.TotalDebitsForDay+m.Amount)-m.DailyLimit),
 	)
 }
+
+// Validate returns a non-nil error if the message is invalid.
+func (m DailyDebitLimitConsumed) Validate() error {
+	if m.TransactionID == "" {
+		return errors.New("DailyDebitLimitConsumed needs a valid transaction ID")
+	}
+	if m.AccountID == "" {
+		return errors.New("DailyDebitLimitConsumed needs a valid account ID")
+	}
+	if m.DebitType == "" {
+		return errors.New("DailyDebitLimitConsumed needs a valid debit type")
+	}
+	if m.Amount < 1 {
+		return errors.New("DailyDebitLimitConsumed needs a valid amount")
+	}
+	if !validation.IsValidBusinessDate(m.Date) {
+		return errors.New("DailyDebitLimitConsumed needs a valid date")
+	}
+	if m.TotalDebitsForDay < 1 {
+		return errors.New("DailyDebitLimitConsumed needs a valid total debits for day")
+	}
+	if m.DailyLimit < 0 {
+		return errors.New("DailyDebitLimitConsumed needs a valid daily limit")
```

Comment:
```suggestion
		return errors.New("DailyDebitLimitConsumed must have a non-negative daily limit")
```

Reply:
I accidentally worded this one different. Let me know if you prefer your suggestion:
I ended up with this: `DailyDebitLimitConsumed must not have a negative daily limit`

Reply:
Your wording is better.